src/events/games/dto/on-match-type-mode.dto.ts

```diff
@@ -0,0 +1,13 @@
+import { MatchType } from '../../../matches/constants/match-type.enum';
+import { MatchGameMode } from '../../../matches/constants/match-game-mode.enum';
+import { IsEnum, IsNotEmpty } from 'class-validator';
+
+export class OnMatchTypeModeDto {
+  @IsNotEmpty()
+  @IsEnum(MatchType)
+  readonly type!: MatchType;
+
+  @IsNotEmpty()
+  @IsEnum(MatchGameMode)
+  readonly mode!: MatchGameMode;
```

Comment:
```suggestion
export interface OnMatchTypeModeDto {
  readonly type!: MatchType;

  readonly mode!: MatchGameMode;
```

이런형태나

`{ type: string; mode: string }` 이런식의 타입도 괜찮을것같습니다.

Reply:
제안해주신 인터페이스 형태로 변경했습니다.